src/controllers/controller.bar.js

```diff
@@ -159,21 +166,27 @@ module.exports = function(Chart) {
 		},
 
 		calculateBarWidth: function(ruler) {
-			var xScale = this.getScaleForId(this.getMeta().xAxisID);
+			var me = this;
+			var meta = me.getMeta();
+			var xScale = me.getScaleForId(meta.xAxisID);
 			if (xScale.options.barThickness) {
 				return xScale.options.barThickness;
 			}
-			return xScale.options.stacked ? ruler.categoryWidth : ruler.barWidth;
+			return (xScale.options.stacked && meta.stack === undefined) ? ruler.categoryWidth : ruler.barWidth;
```

Comment:
That `&& meta.stack === undefined` was to retain the existing behaviour for (non-grouped) stacked bar widths - which are currently independent of barPercentage (and depend only on categoryPercentage). 

I think it would be consistent to have non grouped stacked bar widths proportional to categoryPercentage AND barPercentage. If it's ok to change to make this change, I can remove `meta.stack === undefined` from `getBarCount` and remove the entire `(xScale.options.stacked && meta.stack === undefined) ? ruler.categoryWidth : ` part from here (i.e. just use ruler.barWidth which I believe is already correctly calculated)